### ISSUE TITLE
Use enabledByDefault = false for stability rules instead of lint.xml options

### ISSUE THREAD

**Issue by mcmilj** [slackhq/compose-lints#554]
We currently use a convention plugin to configure linting for modules. This allows us to configure linting from one place.

If the off-by-default stability issues (`ComposeUnstableReceiver`, `ComposeMutableParameters`, and `ComposeUnstableCollections`) were created with `enabledByDefault = false` then we could enable the optional rules in our convention plugin.

Otherwise, we have to create a `lint.xml` file in a project which doesn't otherwise need one, and therefore distribute the linting configuration across multiple files.

See the `LintImplPsiEquals` rule from `com.android.tools.lint:lint-checks` as an example.

Then we could opt-in to the rules like so:
```kt
lint {
    enable += "ComposeUnstableReceiver"
    enable += "ComposeMutableParameters"
    enable += "ComposeUnstableCollections"
}
```

**Comment by ZacSweers** [comment]
Good idea!
